app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailPlayerCrasher.java

```diff
@@ -0,0 +1,179 @@
+package org.schabi.newpipe.fragments.detail;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.ContextThemeWrapper;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.RendererCapabilities;
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.databinding.ListRadioIconItemBinding;
+import org.schabi.newpipe.databinding.SingleChoiceDialogViewBinding;
+import org.schabi.newpipe.player.Player;
+import org.schabi.newpipe.util.ThemeHelper;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+/**
+ * Outsourced logic for crashing the player in the {@link VideoDetailFragment}.
+ */
+public class VideoDetailPlayerCrasher {
+
+    // This has to be <= 23 chars on devices running Android 7 or lower (API <= 25)
+    // or it fails with an IllegalArgumentException
+    // https://stackoverflow.com/a/54744028
+    private static final String TAG = "VideoDetPlayerCrasher";
+
+    @NonNull
+    private final Supplier<Context> contextSupplier;
+    @NonNull
+    private final Supplier<LayoutInflater> layoutInflaterSupplier;
+
+    public VideoDetailPlayerCrasher(
+            @NonNull final Supplier<Context> contextSupplier,
+            @NonNull final Supplier<LayoutInflater> layoutInflaterSupplier
+    ) {
+        this.contextSupplier = contextSupplier;
+        this.layoutInflaterSupplier = layoutInflaterSupplier;
+    }
+
+    private static Map<String, Supplier<ExoPlaybackException>> getExceptionTypes() {
+        final String defaultMsg = "Dummy";
+        final Map<String, Supplier<ExoPlaybackException>> exceptionTypes = new LinkedHashMap<>();
```

Comment:
What about making this a static field in the class instead of generating it each time?